plugins/hdfs_assetstore/plugin_tests/assetstore_test.py

```diff
@@ -0,0 +1,387 @@
+                'size': len(chunk1) + len(chunk2),
+                'mimeType': 'text/plain'
+            })
+        self.assertStatusOk(resp)
+        upload = resp.json
+        absPath = os.path.join(
+            _mockRoot, assetstore['hdfs']['path'][1:],
+            upload['hdfs']['path'])
+
+        # Temp file should be created
+        self.assertEqual(upload['assetstoreId'], str(assetstore['_id']))
+        self.assertFalse(upload['hdfs'].get('imported', False))
+        self.assertTrue(os.path.isfile(absPath))
+
+        # Test uploading to this assetstore
+        @httmock.all_requests
+        def webHdfsMock(url, request):
+            if '50070' in url.netloc:
+                # First request, we must issue a redirect to data node
+                return {
+                    'status_code': 307,
+                    'headers': {
+                        'Location': 'http://localhost:50075/foo'
+                    }
+                }
+            elif url.netloc == 'localhost:50075':
+                # Write the contents
+                with open(absPath, 'a') as f:
+                    f.write(request.body.read())
+                return {
+                    'status_code': 200
+                }
+            else:
+                raise Exception('Unexpected request: ' + repr(url))
+
+        with httmock.HTTMock(webHdfsMock):
+            # Upload first chunk
+            fields = [('offset', 0), ('uploadId', upload['_id'])]
+            files = [('chunk', 'testUpload.txt', chunk1)]
+            resp = self.multipartRequest(
+                path='/file/chunk', user=self.admin, fields=fields, files=files)
+            self.assertStatusOk(resp)
+
+            # Get the offset (simulating resume)
+            upload = resp.json
+            self.assertEqual(upload['received'], len(chunk1))
+            resp = self.request('/file/offset', user=self.admin, params={
+                'uploadId': upload['_id']
+            })
+            self.assertStatusOk(resp)
+            self.assertEqual(resp.json['offset'], len(chunk1))
+
+            fields = [('offset', len(chunk1)), ('uploadId', upload['_id'])]
+            files = [('chunk', 'testUpload.txt', chunk2)]
+            resp = self.multipartRequest(
+                path='/file/chunk', user=self.admin, fields=fields, files=files)
+            self.assertStatusOk(resp)
+            file = resp.json
+
+        # Download the file
+        resp = self.request('/file/{}/download'.format(file['_id']),
+                            isJson=False, user=self.admin)
+        self.assertStatusOk(resp)
+        self.assertEqual(resp.collapse_body(), chunk1 + chunk2)
+
+        # Delete the file, should remove the underlying file on disk
+        resp = self.request('/file/' + str(file['_id']), method='DELETE',
+                            user=self.admin)
+        self.assertEqual(None, self.model('file').load(file['_id']))
+        self.assertFalse(os.path.exists(absPath))
```

Comment:
Is there a convenient way to run the mock server so that I can use it from the user interface?  For instance, the mock_s3.py file can be run in a stand-alone mode to facilitate human testing.